pkg/controllers/util/federatedclient/podinformer.go

```diff
@@ -0,0 +1,123 @@
+package federatedclient
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/sync/semaphore"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/informers"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+func addPodInformer(ctx context.Context,
+	informer informers.SharedInformerFactory,
+	client kubeclient.Interface,
+	podListerSemaphore *semaphore.Weighted,
+	enableSimplifying bool) {
+	informer.InformerFor(&corev1.Pod{}, func(k kubeclient.Interface, resyncPeriod time.Duration) cache.SharedIndexInformer {
+		return cache.NewSharedIndexInformer(
+			podListerWatcher(ctx, client, podListerSemaphore, enableSimplifying),
+			&corev1.Pod{},
+			resyncPeriod,
+			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		)
+	})
+}
+
+func podListerWatcher(ctx context.Context, client kubeclient.Interface, semaphore *semaphore.Weighted, enableSimplifying bool) cache.ListerWatcher {
+	return &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			if semaphore != nil {
+				if err := semaphore.Acquire(ctx, 1); err != nil {
+					return nil, err
+				}
+				defer semaphore.Release(1)
+			}
+			pods, err := client.CoreV1().Pods(corev1.NamespaceAll).List(ctx, options)
+			if err != nil {
+				return nil, err
+			}
+			if enableSimplifying {
+				for i := range pods.Items {
+					simplifyPod(&pods.Items[i])
+				}
+			}
+			return pods, nil
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			watcher, err := client.CoreV1().Pods(corev1.NamespaceAll).Watch(ctx, options)
+			if err != nil {
+				return nil, err
+			}
+			if !enableSimplifying {
+				return watcher, err
+			}
+
+			// It's easy for a consumer to add buffering via an extra
+			// goroutine/channel, but impossible for them to remove it,
+			// so nonbuffered is better. -- from watch.NewStreamWatcher
+			proxyCh := make(chan watch.Event)
+			proxyWatcher := watch.NewProxyWatcher(proxyCh)
+			go func() {
+				defer watcher.Stop()
+				// Closing proxyCh will notify the reflector to stop the current
+				// watching cycle and then restart the list and watch.
+				defer close(proxyCh)
+				for {
+					select {
+					case <-proxyWatcher.StopChan():
+						return
+					case event, ok := <-watcher.ResultChan():
+						if !ok {
+							// the watcher has been closed, stop the proxy
+							return
+						}
+						switch event.Type {
+						case watch.Modified:
+							// Ignore the MODIFIED event, because the request and limit cannot
```

Comment:
This comment only applied to resource collection. With this implementation, the pod informer can be used for other purposes like auto-migration (#9) and resource compensation (#21) which used to rely on pod listing due to the absence of a memory-efficient pod caching mechanism. Meanwhile, handling update events should only increase the CPU but not the memory usage, and the former is currently not a concern.
Is there any other reason for ignoring update events? If not, I think it's ok to handle them.